src/module.h

```diff
@@ -5,10 +5,15 @@
 #include <tuple>
 #include <vector>
 
+class Module;
+
+// TODO: Consider whether these would be more readable as structs
 using specie = std::string;
 using speciesRatios = std::map<specie, int>;
 using reactionRate = int;
 using reaction = std::tuple<speciesRatios, speciesRatios, reactionRate>;
+using speciesMapping = std::map<specie, specie>;
+using composition = std::tuple<Module *, speciesMapping, speciesMapping>;
 
```

Comment:
So they would be structs with 1 line?
To some extent they may be more readable, but i think it is better code, to have them as structs

Reply:
Most of these structs would have multiple lines. Then, if you had a `composition c`, you could access its members with `c.inputs` instead of `std::get<1>(c)`

Reply:
i think that would make it a **lot** more readable tbh

Reply:
I'll promote it to an issue then